### PR TITLE
kafka sink: support partition by rowid or unique value

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
support partition by rowid or unique value

[DMN] we should merge this pr after testing.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/entry/schema_storage.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ type TableInfo struct {
 	IndicesOffset map[int64]int
 	UniqueColumns map[int64]struct{}
 	handleColID   int64
+	IndieMarkCol  string
```
Comment (amyangfei) [inline cdc/entry/schema_storage.go]: What is `IndieMarkCol` short for, better to add some comment

File: cdc/entry/schema_storage.go [hunk 2]
```diff
@@ -105,6 +110,17 @@ func WrapTableInfo(info *timodel.TableInfo) *TableInfo {
 				ti.UniqueColumns[ti.Columns[col.Offset].ID] = struct{}{}
 			}
 		}
+		if idx.Primary || idx.Unique {
```
Comment (amyangfei) [inline cdc/entry/schema_storage.go]: Is `primary key` be counted twice, as L104 has increased once.

Comment (leoppro) [reply cdc/entry/schema_storage.go]: no, if pk is handle, `primary index` will be not included in `ti.Indices`.

**Pull request merged**